qa/rpc-tests/test_framework/test_framework.py

```diff
@@ -183,12 +179,41 @@ def main(self):
                     from collections import deque
                     print("".join(deque(open(f), MAX_LINES_TO_PRINT)))
         if success:
-            print("Tests successful")
+            self.log.info("Tests successful")
             sys.exit(0)
         else:
-            print("Failed")
+            self.log.warning("Failed")
             sys.exit(1)
 
+    def start_logging(self):
+        # Add logger and logging handlers
+        self.log = logging.getLogger('TestFramework')
+        self.log.setLevel(logging.DEBUG)
+        # Create file handler to log all messages
+        fh = logging.FileHandler(self.options.tmpdir + '/test_framework.log')
+        fh.setLevel(logging.DEBUG)
+        # Create console handler to log messages to stderr. By default this logs only error messages, but can be configured with --loglevel.
+        ch = logging.StreamHandler()
+        console_log_level = getattr(logging, self.options.loglevel.upper(), None)
```

Comment:
Agree. I was following the Python 'logging how to' guide here: https://docs.python.org/3/howto/logging.html#logging-to-a-file which uses getattr(). I guess that was written before 3.2, which is when setlevel() was updated to take a string argument.

I've tested the logging module and:
- we do need to use upper() to set the log levels to uppercase
- calling with an invalid string causes logging to raise a helpful exception (I was worried that it might fail silently).

So calling setlevel directly with self.options.loglevel.upper() is better.